app/models/sample_type.rb

```diff
@@ -156,7 +162,18 @@ def self.can_create?
     can && (!Seek::Config.project_admin_sample_type_restriction || User.current_user.is_admin_or_project_administrator?)
   end
 
+  def can_edit?(user = User.current_user)
+    super && !locked?
+  end
+
+  def can_manage?(user = User.current_user)
+    super && !locked?
+  end
+
   def can_delete?(user = User.current_user)
+    # Should not be able to delete a sample type if it is locked
+    return false if locked?
```

Comment:
I think these should be done using the `state_allows_edit?` etc. checks

e.g. https://github.com/seek4science/seek/blob/0995d8634d2da1ae0ac77851ecd36a467f94d25d/app/models/assay.rb#L136-L138